dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/workflow/oracle/xmlworkflow/V6.0_2015.08.11__DS-2701_Xml_Workflow_Migration.sql

```diff
@@ -18,11 +18,65 @@ UPDATE collection SET workflow_step_3 = null;
 ALTER TABLE cwf_workflowitem RENAME COLUMN item_id to item_legacy_id;
 ALTER TABLE cwf_workflowitem ADD item_id RAW(16) REFERENCES Item(uuid);
 UPDATE cwf_workflowitem SET item_id = (SELECT item.uuid FROM item WHERE cwf_workflowitem.item_legacy_id = item.item_id);
+DECLARE
+  COUNT_INDEXES INTEGER;
+  INDEX_NAME VARCHAR(256);
+  TABLE_NAME_D VARCHAR(256) := 'cwf_workflowitem';
+  COLUMN_NAME_D VARCHAR(256) := 'item_legacy_id';
+BEGIN
+  select count(c.INDEX_NAME) into COUNT_INDEXES
+  from USER_INDEXES i, USER_IND_COLUMNS c
+  where i.TABLE_NAME = upper(TABLE_NAME_D)
+        and i.UNIQUENESS = 'UNIQUE'
+        and i.TABLE_NAME = c.TABLE_NAME
+        and i.INDEX_NAME = c.INDEX_NAME
+        and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+
+  IF COUNT_INDEXES > 0 THEN
+    select c.INDEX_NAME into INDEX_NAME
+    from USER_INDEXES i, USER_IND_COLUMNS c
+    where i.TABLE_NAME = upper(TABLE_NAME_D)
+          and i.UNIQUENESS = 'UNIQUE'
+          and i.TABLE_NAME = c.TABLE_NAME
+          and i.INDEX_NAME = c.INDEX_NAME
+          and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+    EXECUTE IMMEDIATE 'ALTER TABLE ' || TABLE_NAME_D || ' DROP constraint ' || INDEX_NAME;
+  END IF;
+END;
+/
+
 ALTER TABLE cwf_workflowitem DROP COLUMN item_legacy_id;
 
 ALTER TABLE cwf_workflowitem RENAME COLUMN collection_id to collection_legacy_id;
 ALTER TABLE cwf_workflowitem ADD collection_id RAW(16) REFERENCES Collection(uuid);
 UPDATE cwf_workflowitem SET collection_id = (SELECT collection.uuid FROM collection WHERE cwf_workflowitem.collection_legacy_id = collection.collection_id);
+DECLARE
+  COUNT_INDEXES INTEGER;
+  INDEX_NAME VARCHAR(256);
+  TABLE_NAME_D VARCHAR(256) := 'cwf_workflowitem';
+  COLUMN_NAME_D VARCHAR(256) := 'collection_legacy_id';
+BEGIN
+  select count(c.INDEX_NAME) into COUNT_INDEXES
+  from USER_INDEXES i, USER_IND_COLUMNS c
+  where i.TABLE_NAME = upper(TABLE_NAME_D)
+        and i.UNIQUENESS = 'UNIQUE'
+        and i.TABLE_NAME = c.TABLE_NAME
+        and i.INDEX_NAME = c.INDEX_NAME
+        and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+
+  IF COUNT_INDEXES > 0 THEN
+    select c.INDEX_NAME into INDEX_NAME
+    from USER_INDEXES i, USER_IND_COLUMNS c
+    where i.TABLE_NAME = upper(TABLE_NAME_D)
+          and i.UNIQUENESS = 'UNIQUE'
+          and i.TABLE_NAME = c.TABLE_NAME
+          and i.INDEX_NAME = c.INDEX_NAME
+          and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+    EXECUTE IMMEDIATE 'ALTER TABLE ' || TABLE_NAME_D || ' DROP constraint ' || INDEX_NAME;
+  END IF;
+END;
+/
+
```

Comment:
Same as above. I wonder if this constraint can be dropped just before this existing script triggers.

(There's many more of these same DROP constraint SQL statements in this same file that could be handled similarly. I won't repeat this comment for all of them, but I feel they all may fall into the same situation)